### PR TITLE
Add a preview for the "next release" change notes

### PR DESCRIPTION
**Pull request by webknjaz**
# Summary

This change integrates `sphinxcontrib-towncrier` into the docs build.
It uses Towncrier and its configs to render a draft changelog as RST
and injects that into the `changelog.rst` document right above the
older changelog entries using `towncrier-draft-entries` directive.

I figured, it'd be nicer to be able to have a sneak-peak into the future
release changes. As an alternative to navigating to a folder with a
bunch of standalone uncategorized files.

Usage in the wild: https://pip.pypa.io/en/latest/news/#not-yet-released-2022-08-21.

Preview: https://attrs--1016.org.readthedocs.build/en/1016/changelog.html#changelog.

### CODE REVIEW

File: changelog.d/1016.doc.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Render the unreleased ``attrs`` version change notes on the changelog page in docs.
```
Comment (hynek) [inline changelog.d/1016.doc.rst:1]: We don’t have a doc category, so this will have to do without a fragment.

File: docs/changelog.rst [hunk 2]
```diff
@@ -1 +1,7 @@
 .. include:: ../CHANGELOG.rst
+   :end-before: Changes for the upcoming release can be found
+
+.. towncrier-draft-entries:: |release|:sub:`/UNRELEASED DRAFT/`
```
Comment (hynek) [inline docs/changelog.rst:4]: What exactly is the `sub` supposed to do? I would have guessed substitute the release but it looks like it uses a `<sub>` tag or something? Is that on purpose? If yes: why? :)

**Review by hynek — CHANGES_REQUESTED** [review]
Looks cool, but please address my comments. :)

File: changelog.d/1016.doc.rst (hunk 1 above)
Comment (webknjaz) [reply changelog.d/1016.doc.rst:1]: Oh, I didn't realize that. That's a power of habit :)

Did you consider including `CONTRIBUTING.md` into the Sphinx docs? It's pretty straightforward to add a single Markdown document there via MyST. No need to convert everything into the same format.

P.S. Feel free to try out https://github.com/apps/psf-chronographer if you want to verify the presense of the allowed change note types in PRs, like aiohttp does.

File: docs/changelog.rst (hunk 2 above)
Comment (webknjaz) [reply docs/changelog.rst:4]: Would you prefer something like this?
```suggestion
.. towncrier-draft-entries:: |release| | *UNRELEASED DRAFT*
```
Any other thoughts?

Comment (hynek) [reply docs/changelog.rst]: maybe even remove `|release|` too? pip hasn't it either

Comment (hynek) [reply docs/changelog.rst]: fee free to use their `Not yet released`

File: changelog.d/1016.doc.rst (hunk 1 above)
Comment (webknjaz) [reply changelog.d/1016.doc.rst]: Personally, I was a little lost when I couldn't locate this in the docs, among the top-level items. I think there was a link on one of the pages where I never land coming from Google.

File: docs/changelog.rst (hunk 2 above)
Comment (webknjaz) [reply docs/changelog.rst]: Yeah, maybe even `Will be included in the upcoming release`?

Comment (hynek) [reply docs/changelog.rst]: naw too long. what's good enough for pip, is good enough for us (in this case)

Comment (webknjaz) [reply docs/changelog.rst]: I've gone for what pip has.

File: changelog.d/1016.doc.rst (hunk 1 above)
Comment (hynek) [reply changelog.d/1016.doc.rst]: there's one on the github readme, the main docs page, and the pypi page :D

maybe i'll just add it to the list so it's less embedded

**Review by hynek — APPROVED** [review]
Sweet, thanks!

**Pull request merged**